src/Arrays/MPIStateArrays.jl

```diff
@@ -856,4 +856,33 @@ end
     end
 end
 
+"""
+    throw_norm_error(Q::MPIStateArray)
+
+Call if `isnan(norm(Q)) == true` to provide a
+detailed error of which field(s) have NaNs.
+"""
+function throw_norm_error(Q::MPIStateArray)
+    vs = vars(Q)
+    nan_fields = []
+    Qarr = Array(Q)
+    for ftc in flattened_tup_chain(vs)
+        i_vars = varsindex(vs, wrap_val.(ftc)...)
+        if length(i_vars) > 1 # array
+            for (j, i_var) in enumerate(i_vars)
+                if isnan(norm(Qarr[:, :, i_var]))
```

Comment:
Good point. Also, I'm indexing into the wrong rank 🤦🏻 . Fixing a bunch of things and adding tests.

Reply:
The problem with using the `MPIStateArray` `norm`, though, is that if the data is on the GPU, I'm not sure how we can `getindex` for the specific field.

Reply:
I think you **don't** want the `MPIStateArray` `norm`.

Reply:
Wait, wouldn't we want the MPIStateArray norm so that it works on multi-node?

Reply:
We want to use that one on the outside, in the primary check, yes. Now that I think about it, my concern is misplaced -- I was worried that not all nodes would see the `NaN` from the primary `norm` call so they would enter different code paths (and a subsequent MPI collective could block forever), but that's not correct -- if there's a `NaN`, all nodes will see it.

So you're right, you probably want to use the `MPIStateArray` `norm` on this inner check as well otherwise it will be local only, which means it is possible that only some of the nodes will find the `NaN`.

To do so, I think you have to use the `dims` parameter to `norm` rather than indexing into `Qarr`.

Reply:
That means that we'd need to fix/merge #590. That seems to be much more work than I set out for. Going to stick with a single rank error for now so that we get some utility out of this.

Reply:
Actually, I think we can just use `mapreduce` with `dims`. Fixing it up now

Reply:
Ok, updated things